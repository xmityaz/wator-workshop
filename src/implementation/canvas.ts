import {OceanSize, PetMap} from '../types';

export const PET_SIZE = 5;
export const FISH_COLOR = '#ff0000';
export const SHARK_COLOR = '#0000ff';

export const drawPetMap = (petMap: PetMap, oceanSize: OceanSize, ctx: CanvasRenderingContext2D) => {};
