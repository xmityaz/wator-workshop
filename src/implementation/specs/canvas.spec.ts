import {drawPetMap, PET_SIZE} from '../canvas';
import {aFish, aShark} from './testUtils';

describe('canvas', () => {
  let canvas: HTMLCanvasElement;
  const oceanSize = {width: 3, height: 3};

  beforeEach(() => {
    canvas = document.createElement('canvas');
    canvas.height = oceanSize.height * PET_SIZE;
    canvas.width = oceanSize.width * PET_SIZE;
  });

  describe('drawPetMap', () => {
    it('case 1 ', () => {
      const ctx = canvas.getContext('2d') as CanvasRenderingContext2D;
      const petMap = [
        [null, aFish(1), null],
        [null, aFish(1), null],
        [null, aFish(1), null]
      ];

      drawPetMap(petMap, oceanSize, ctx);

      expect(ctx.__getDrawCalls()).toMatchSnapshot();
    });

    it('case 2', () => {
      const ctx = canvas.getContext('2d') as CanvasRenderingContext2D;
      const petMap = [
        [aFish(1), aShark(1, 1), aFish(1)],
        [aShark(1, 1), aFish(1), aShark(1, 1)],
        [aFish(1), aShark(1, 1), aFish(1)]
      ];

      drawPetMap(petMap, oceanSize, ctx);

      expect(ctx.__getDrawCalls()).toMatchSnapshot();
    });

    it('case 3', () => {
      const ctx = canvas.getContext('2d') as CanvasRenderingContext2D;
      const petMap = [
        [aShark(1, 1), aShark(1, 1), aShark(1, 1)],
        [aShark(1, 1), null, aShark(1, 1)],
        [null, aShark(1, 1), null]
      ];

      drawPetMap(petMap, oceanSize, ctx);

      expect(ctx.__getDrawCalls()).toMatchSnapshot();
    });

    it('case 4', () => {
      const ctx = canvas.getContext('2d') as CanvasRenderingContext2D;
      const petMap = [
        [null, null, null],
        [null, null, null],
        [null, null, null]
      ];

      drawPetMap(petMap, oceanSize, ctx);

      expect(ctx.__getDrawCalls()).toMatchSnapshot();
    });
  });
});
