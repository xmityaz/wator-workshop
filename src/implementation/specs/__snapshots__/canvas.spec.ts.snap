// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`canvas drawPetMap case 1  1`] = `
Array [
  Object {
    "props": Object {
      "height": 15,
      "width": 15,
      "x": 0,
      "y": 0,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  Object {
    "props": Object {
      "height": 5,
      "width": 5,
      "x": 5,
      "y": 0,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  Object {
    "props": Object {
      "height": 5,
      "width": 5,
      "x": 5,
      "y": 5,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  Object {
    "props": Object {
      "height": 5,
      "width": 5,
      "x": 5,
      "y": 10,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
]
`;

exports[`canvas drawPetMap case 2 1`] = `
Array [
  Object {
    "props": Object {
      "height": 15,
      "width": 15,
      "x": 0,
      "y": 0,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  Object {
    "props": Object {
      "height": 5,
      "width": 5,
      "x": 0,
      "y": 0,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  Object {
    "props": Object {
      "height": 5,
      "width": 5,
      "x": 0,
      "y": 5,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  Object {
    "props": Object {
      "height": 5,
      "width": 5,
      "x": 0,
      "y": 10,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  Object {
    "props": Object {
      "height": 5,
      "width": 5,
      "x": 5,
      "y": 0,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  Object {
    "props": Object {
      "height": 5,
      "width": 5,
      "x": 5,
      "y": 5,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  Object {
    "props": Object {
      "height": 5,
      "width": 5,
      "x": 5,
      "y": 10,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  Object {
    "props": Object {
      "height": 5,
      "width": 5,
      "x": 10,
      "y": 0,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  Object {
    "props": Object {
      "height": 5,
      "width": 5,
      "x": 10,
      "y": 5,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  Object {
    "props": Object {
      "height": 5,
      "width": 5,
      "x": 10,
      "y": 10,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
]
`;

exports[`canvas drawPetMap case 3 1`] = `
Array [
  Object {
    "props": Object {
      "height": 15,
      "width": 15,
      "x": 0,
      "y": 0,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  Object {
    "props": Object {
      "height": 5,
      "width": 5,
      "x": 0,
      "y": 0,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  Object {
    "props": Object {
      "height": 5,
      "width": 5,
      "x": 0,
      "y": 5,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  Object {
    "props": Object {
      "height": 5,
      "width": 5,
      "x": 5,
      "y": 0,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  Object {
    "props": Object {
      "height": 5,
      "width": 5,
      "x": 5,
      "y": 10,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  Object {
    "props": Object {
      "height": 5,
      "width": 5,
      "x": 10,
      "y": 0,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  Object {
    "props": Object {
      "height": 5,
      "width": 5,
      "x": 10,
      "y": 5,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
]
`;

exports[`canvas drawPetMap case 4 1`] = `
Array [
  Object {
    "props": Object {
      "height": 15,
      "width": 15,
      "x": 0,
      "y": 0,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
]
`;
